data_utils.py

```diff
@@ -63,118 +62,76 @@ def get_dataset_loader(config, dataset, shuffle=False, train=True):
-    return dataset
+def tokenize(text):
+    tokenizer = RegexpTokenizer(r'\w+')
+    return [t.lower() for t in tokenizer.tokenize(text) if not t.isnumeric()]
 
 
-@log.enter('load_dataset')
-def load_dataset(config):
-    """Preparing TextDataset from raw data."""
-    data_dir = os.path.join(config['data_dir'], config['data_name'])
-    os.makedirs(data_dir, exist_ok=True)
-
-    # Load from cache if exists, otherwise load raw data and tokenize
-    cache_path = os.path.join(data_dir, 'cache.pkl')
-    if os.path.exists(cache_path):
-        log.info(f'Load existing cache from {cache_path}.')
-        with open(cache_path, 'rb') as fp:
-            datasets = pickle.load(fp)
-    else:
-        datasets = _load_raw_data(config, data_dir)
-        for split in datasets.keys():
-            datasets[split] = _preprocess_on_split(datasets[split])
-        with open(cache_path, 'wb') as fp:
-            pickle.dump(datasets, fp)
-
-    map_path = os.path.join(data_dir, f'vocab_label_map_{config["min_vocab_freq"]}.pkl')
-    map_path = config['vocab_label_map'] or map_path
-    if map_path and os.path.exists(map_path):
-        log.info(f'Load existing Vocab and classes from {map_path}.')
-        with open(map_path, 'rb') as fp:
-            text_dict, classes = pickle.load(fp)
-    else:
-        classes = set()
-        for dataset in datasets.values():
-            for d in tqdm.tqdm(dataset):
-                classes.update(d['label'])
-        classes = sorted(classes)
-
-        text_dict = build_text_dict(datasets['train'], config['min_vocab_freq'], config['vocab_file'])
-        with open(map_path, 'wb') as fp:
-            pickle.dump((text_dict, classes), fp)
-        log.info(f'Save {map_path}')
+def _load_raw_data(path, is_test=False):
+    log.info(f'Load data from {path}.')
+    data = pd.read_csv(path, sep='\t', names=['label', 'text'],
+                       converters={'label': lambda s: s.split(),
+                                   'text': tokenize})
+    data = data.reset_index().to_dict('records')
+    if not is_test:
+        data = [d for d in data if len(d['label']) > 0]
+    return data
 
-    if 'dev' not in datasets:
-        dev_size = config['dev_size'] if config['dev_size'].is_integer() else int(config['dev_size'] * len(datasets['train']))
-        train_size = len(datasets['train']) - dev_size
-        datasets['train'], datasets['dev'] = torch.utils.data.random_split(datasets['train'], [train_size, dev_size], generator=torch.Generator().manual_seed(42))
 
-    for split in datasets.keys():
-        datasets[split] = TextDataset(datasets[split], text_dict, classes, config['max_seq_length'])
+@log.enter('load_dataset')
+def load_datasets(config):
+    datasets = {}
+    test_path = config.test_path or os.path.join(config.data_dir, 'test.txt')
+    if os.path.exists(test_path):
+        datasets['test'] = _load_raw_data(test_path, is_test=True)
+    if config.eval:
```

Comment:
what if config.eval is true but test_path does not exist? Need an error message

Reply:
Fixed.